modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm

```diff
@@ -23,68 +23,233 @@
+			var/mob/living/carbon/victim = hemophage.pulling
+			if(hemophage.blood_volume >= BLOOD_VOLUME_MAXIMUM)
+				to_chat(hemophage, span_warning("You're already full!"))
+				return
+
+			if(victim.stat == DEAD)
+				to_chat(hemophage, span_warning("You need a living victim!"))
+				return
+
+			if(!victim.blood_volume || (victim.dna && ((NOBLOOD in victim.dna.species.species_traits) || victim.dna.species.exotic_blood)))
+				to_chat(hemophage, span_warning("[victim] doesn't have blood!"))
+				return
+
+
+			var/blood_volume_difference = BLOOD_VOLUME_MAXIMUM - hemophage.blood_volume //How much capacity we have left to absorb blood
+			// We start by checking that the victim is a human and they have a client, so we can give them the
+			// beneficial status effect for drinking higher-quality blood.
+			var/is_target_human_with_client = istype(victim, /mob/living/carbon/human) && victim.client
+
+			if(ismonkey(victim))
+				if(hemophage.blood_volume >= BLOOD_VOLUME_NORMAL)
+					to_chat(hemophage, span_warning("Their inferior blood cannot sate you any further!"))
+					return
+
+				is_target_human_with_client = FALSE // Sorry, not going to get the status effect from monkeys, even if they have a client in them.
+
+				blood_volume_difference = BLOOD_VOLUME_NORMAL - hemophage.blood_volume
+
+
+			COOLDOWN_START(tongue, drain_cooldown, 3 SECONDS)
+			if(victim.can_block_magic(MAGIC_RESISTANCE_HOLY, charge_cost = 0))
+				victim.show_message(span_warning("[hemophage] tries to bite you, but stops before touching you!"))
+				to_chat(hemophage, span_warning("[victim] is blessed! You stop just in time to avoid catching fire."))
+				return
+
+			if(victim.has_reagent(/datum/reagent/consumable/garlic))
+				victim.show_message(span_warning("[hemophage] tries to bite you, but recoils in disgust!"))
+				to_chat(hemophage, span_warning("[victim] reeks of garlic! you can't bring yourself to drain such tainted blood."))
+				return
+
+			if(!do_after(hemophage, 3 SECONDS, target = victim))
+				return
+
+			var/drained_blood = min(victim.blood_volume, HEMOPHAGE_DRAIN_AMOUNT, blood_volume_difference)
+
+			victim.show_message(span_danger("[hemophage] drains some of your blood!"))
+			to_chat(hemophage, span_notice("You drink some blood from [victim]![is_target_human_with_client ? " That tasted particularly good!" : ""]"))
+			playsound(hemophage, 'sound/items/drink.ogg', 30, TRUE, -2)
+
+			victim.blood_volume = clamp(victim.blood_volume - drained_blood, 0, BLOOD_VOLUME_MAXIMUM)
+			hemophage.blood_volume = clamp(hemophage.blood_volume + drained_blood, 0, BLOOD_VOLUME_MAXIMUM)
+
+			log_combat(hemophage, victim, "drained [drained_blood]u of blood from", addition = " (NEW BLOOD VOLUME: [victim.blood_volume] cL)")
+
+			if(is_target_human_with_client)
+				hemophage.apply_status_effect(/datum/status_effect/quality_blood_satiety)
+
+			if(!victim.blood_volume)
+				to_chat(hemophage, span_notice("You finish off [victim]'s blood supply."))
```

Comment:
balloon alert?

Reply:
I disagree with this one, this kind of stuff needs to be something that stays in your chat, and that you can't miss, because it's a very important thing.

Mind you, it's nearly impossible to get there to begin with, I believe the only ways are either Godmode or being in stasis, and even in stasis I'm not entirely sure you can survive being below the instadeath blood threshold.